ResearchKit/Common/ORKSurveyAnswerCellForLocation.m

```diff
@@ -0,0 +1,128 @@
+}
+
+- (void)setUpConstraints {
+    NSMutableArray *constraints = [NSMutableArray new];
+
+    NSDictionary *views = NSDictionaryOfVariableBindings(_selectionView);
+    ORKEnableAutoLayoutForViews([views allValues]);
+    
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_selectionView]|"
+                                                                 options:NSLayoutFormatDirectionLeadingToTrailing
+                                                                 metrics:nil
+                                                                   views:views]];
+    [constraints addObjectsFromArray:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|[_selectionView]|"
+                                                                 options:NSLayoutFormatDirectionLeadingToTrailing
+                                                                 metrics:nil
+                                                                   views:views]];
+    
+    NSLayoutConstraint *resistsCompressingMapConstraint = [NSLayoutConstraint constraintWithItem:_selectionView
+                                                                                       attribute:NSLayoutAttributeWidth
+                                                                                       relatedBy:NSLayoutRelationGreaterThanOrEqual
+                                                                                          toItem:nil
+                                                                                       attribute:NSLayoutAttributeNotAnAttribute
+                                                                                      multiplier:1.0
+                                                                                        constant:20000.0];
+    resistsCompressingMapConstraint.priority = UILayoutPriorityDefaultHigh;
+    [constraints addObject:resistsCompressingMapConstraint];
+    
+    [NSLayoutConstraint activateConstraints:constraints];
+}
+
+- (void)answerDidChange {
+    id answer = self.answer;
+    
+    id displayValue = (answer && answer != ORKNullAnswerValue()) ? answer : nil;
+    if ([displayValue isKindOfClass:[MKPlacemark class]]) {
+        _selectionView.answer = [[ORKPlacemark alloc] initWithPlacemark:(MKPlacemark *)answer];
+    }
+    
+    NSString *placeholder = self.step.placeholder? : ORKLocalizedString(@"PLACEHOLDER_TEXT_OR_NUMBER", nil);
+    [_selectionView setPlaceholderText:placeholder];
+}
+
+- (void)locationSelectionViewDidChange:(ORKLocationSelectionView *)view {
+    if (_selectionView.answer != nil) {
+        [self ork_setAnswer:_selectionView.answer];
+    } else {
+        [self ork_setAnswer:ORKNullAnswerValue()];
+    }
+}
+
+- (void)locationSelectionView:(ORKLocationSelectionView *)view didFailWithError:(NSError *)error {
+    [self showValidityAlertWithTitle:ORKLocalizedString(@"LOCATION_ERROR_TITLE", @"") message:error.localizedDescription];
```

Comment:
@YuanZhu-apple What do you suggest? Should I Include a custom localized string for each possible error code?